Area Annotation Toolbox GUI

yinghj opened the pull request:
Deprecate the onGui UI drawing methods for area annotation toolbox, replacing it with UI elements on Canvas in scene.

yinghj committed:
Version 1 of redoing area annotation toolbox to deprecate onGui().
TODO: Scrolling issue if more than 3 entries added. Prefab scaling.

yinghj committed:
Make area annotation UI entry scalable according to resolution.

yinghj committed:
Make the journalButton entry prefab scalable too.

Charlieranderson commented on Project/Assets/Prairie/Framework/Script/Annotation/AreaAnnotationGui.cs:
```diff
@@ -0,0 +1,83 @@
+		gameObject.transform.Find(AREAANNOTATION).gameObject.SetActive (active);
+	}
+
+	public void ActivateGui() {
```
Could you maybe add some comments on these small methods just detailing where and when they might be called? I can't tell just from looking at this code.

Charlieranderson reviewed (commented):
Looks good except for the one change I asked for.

yinghj committed:
Fix bugs caused by recycling prefab. Rollback to the easier approach of destroying and instantiating every time when an annotation entry changed.

yinghj commented:
Also added comments that addressed Charlie's request.

yinghj committed:
Remove the deprecated drawToolbar method for area annotation GUI in first person interactor.

yinghj committed:
Edit the name of the area annotation UI element.

yinghj commented:
Do NOT merge. Discovered new bug when moving from one set of overlapping area annotation colliders to another.

yinghj committed:
Fix continuous entering and exitting overlapping trigger issue for area annotation toolbox UI.
Store corresponding UI entry in each annotation content object.
Change Prairie Player Collision Detection setting to Continuous Dynamic.

Charlieranderson commented:
Is the bug fixed now?

yinghj commented:
Yes. This is ready for review and mergeable now.

adanter reviewed (approved):
Looks good to me
